compiler/infra/ForwardList.hpp

```diff
@@ -0,0 +1,684 @@
+#ifndef FORWARD_LIST_HPP
+#define FORWARD_LIST_HPP
+
+#pragma once
+
+#if 0
```

Comment:
Contained within the `#if 0` section is what is required to implement forward_list as an extension to std::forward_list, and is, ideally, what we would switch to once all our target build environments support using std::forward_list.